File: src/main/java/com/apitest/dataProvider/TestData.java
package com.apitest.dataProvider;

import java.lang.annotation.*;


@Repeatable(TestDatas.class)
public @interface TestData {
    DataSource source() default DataSource.Spring;
    boolean single() default true;
    Class<? extends IDataProvider> dataProvider() default IDataProvider.class;
    boolean parallel() default false;
    String file() default "";
    String pattern() default "";
}

File: src/main/java/com/apitest/dataProvider/TestData.kt
//package com.apitest.dataProvider
//
//import kotlin.reflect.KClass
//
//
//@Retention(AnnotationRetention.RUNTIME)
//@Target(AnnotationTarget.FUNCTION)
//@Repeatable
////todo 目前kotlin不支持repeatable
//annotation class TestData(
//        val source: DataSource = DataSource.Spring,
//        val single:Boolean=true,
//        val dataProvider: KClass<out IDataProvider<*>> = IDataProvider::class,
//        val parallel:Boolean = false,
//        val file:String="",
//        val pattern:String="")
//
//
//
//
//
